server-spi-private/src/main/java/org/keycloak/verifiableclaims/VerifiableClaimProvider.java

```diff
@@ -0,0 +1,36 @@
+package org.keycloak.verifiableclaims;
+
+import org.keycloak.models.*;
+import org.keycloak.provider.Provider;
+import org.keycloak.userprofile.Attributes;
+import org.keycloak.userprofile.UserProfile;
+import org.keycloak.userprofile.UserProfileContext;
+import org.keycloak.verifiableclaims.model.*;
+
+import java.util.Map;
+
+public interface VerifiableClaimProvider extends Provider {
+
+    Map<String, VerifiableAttributeConfig> getVerifiableAttributeConfig(KeycloakSession session, RealmModel realm);
+
+    UpdateDecision onAttributesAboutToUpdate(KeycloakSession session,
```

Comment:
What if we have a `VerifiableClaim` type to abstract the different types of verifiable claims: user attributes, roles, groups, etc?

By doing that, we can make this provider more generic and capable of handling different types of verifiable claims. As it stands today, it is very specific to user profile/attributes.

For instance, the `onAttributesAboutToUpdate` would accept a `VerifiableClaim` object that wraps a user attribute, role grant, or group membership. We could also consider changing this method to accept a "update function" so that it will only be invoked (and the data persisted) if the attribute should be "persisted now".